### PR TITLE
Enforce code style rule IDE0053

### PR DESCRIPTION
**Pull request by RoosterDragon**
Enforce IDE0053 Use expression body for lambdas https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0053

We can configure the style option, have gone for `when_on_single_line` initially.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -123,6 +123,8 @@ dotnet_style_predefined_type_for_locals_parameters_members = true
 
 # Use expression body if a local function is a single line.
 csharp_style_expression_bodied_local_functions = when_on_single_line
+# This rule is buggy and not enforced for builds. ':warning' will at least enforce it in the IDE.
+csharp_style_expression_bodied_lambdas = when_on_single_line:warning
```
Comment (PunkPun) [inline .editorconfig:128]: missing newline. 

Do I understand correctly that make check / GH actions won't detect this violation?

Comment (RoosterDragon) [reply .editorconfig:128]: Wanted it grouped with the other expression body option, fixed the preceeding comment for that.

You are correct. The desired setup (below) fails to report any violations. The IDE-only `:severity` syntax gives us something for now. But as it is IDE-only, command lines builds (and thus CI) won't get them.
```ini
csharp_style_expression_bodied_lambdas = when_on_single_line
dotnet_diagnostic.IDE0053.severity = warning
```

Comment (PunkPun) [reply .editorconfig:128]: Our style is to have a newline before comments. I assume those don't trigger inside this file

Comment (RoosterDragon) [reply .editorconfig:128]: Updated.

**Review by PunkPun — APPROVED** [review]

**Review by penev92 — APPROVED** [review]
Ok.

**Comment by penev92** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/23b0c5a249e6a0eb7bec9dbdbc8b77377f18467b...5f85540f069b712eb4ef99b998808700e9514ea5)

**Pull request merged**